cirq/protocols/__init__.py

```diff
@@ -0,0 +1,4 @@
+from cirq.protocols.unitary import (
+    SupportsUnitary,
+    unitary,
+)
```

Comment:
It would be nice to have some module docs about the intent here. IIUC, the idea is that these "protocols" work like python builtin functions that are implemented by calling special `__methods__` that may or may not be defined on a particular object. We use mypy `Protocol` objects to describe our protocols to mypy and to allow explicit or implicit (ducktyped) implementation of those protocols. Also, we use single-underscoe `_methods_` to distinguish from python's built-in special methods.

Reply:
I don't think this is the right place to put those docs. How would the user end up here? It's far more likely they'd end up in `cirq/__init__.py` (by manual exploration) or in `cirq/protocols/unitary.py` (by go-to-definition).

I do think we will need a place in the documentation labelled "magic methods" or similar where these are all listed.